packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx

```diff
@@ -62,8 +63,18 @@ export const ButtonDownloadRecoveryKey = ({
     { supportURL }
   );
 
-  const fileContent = new Blob(
+  // 'keyWithContext' is currently disabled by default due to an issue with encoding recognition on Android.
+  // Non-latin text (e.g., Hebrew) may be displayed as jibberish due to incorrect encoding detection.
+  // In addition, localized strings contain important directionality markers that are hidden on Mac
+  // but visible on many other devices. On Apple devices, these directionality markers get copied with the key,
+  // and the key is rejected as invalid during password reset.
```

Comment:
What do these markers look like? Is it something we can strip on submit like we do spaces? Doesn't help with the comment about Hebrew though...

What do our localized plaintext emails look like, are they not experiencing these issues? If they aren't, the fix (not in this PR) might be to repurpose our Localizer class to be either used here (sounds like Reino was already thinking about using it client-side), or the logic needed from it pulled and used. We probably need a similar ticket filed if our emails experience the same thing.

Reply:
The markers are rendered differently depending on the device/application. Sometimes invisible, sometimes square boxes, sometimes little symbols with FSI/PDI letters.

We could strip them out on submit or leave the key out of the localized string to resolve [FXA-7571](https://mozilla-hub.atlassian.net/browse/FXA-7571) , but we're left with the issue that there are directionality markers around other bits of text in this download file including brand terms, and that doesn't look great on most devices/applications except Apple's [FXA-7564](https://mozilla-hub.atlassian.net/browse/FXA-7564).

Our plain text emails do include directionality markers, but they're hidden in maildev. Not sure if they might be visible in other email clients. The confirmation codes are fine functionality-wise because they're on their own line and not included as a variable in a localized string --> no directionality marker(s) around the emailed codes.

Reply:
Well, that is a tricky problem. I'm just reiterating the ticket here but looks like Windows is not respecting the directionality markers at all if I'm seeing the screenshots correctly; English shows the unicode control characters and renders LTR (probably default) and RTL languages show the unicode control characters and are still rendering LTR.

Notepad is notorious for not displaying unicode correctly, but it's strange it's also occurring in Notepad++... If we don't use PDFs (unsure if that would solve this), this is not ideal but I wonder if Windows would [interpret "overrides" or "embedding" better than "isolates"](https://www.w3.org/International/questions/qa-bidi-unicode-controls.en.html#basedirection)? I'm not sure if we can parse the text and replace `[U+2068]` with `\u202E` (not 100% sure that's how it'd need to be in code) better than `[U+2068]` or if that would need to happen on the l10n side somehow. 🤔

Reply:
TLDR; Getting encoding right on all devices/apps is very tricky. For the time being, I've reverted the default download to "key only" but included a few options that we can test in staging with a param. Also filed a spike to investigate options https://mozilla-hub.atlassian.net/browse/FXA-7790 because this testing was growing beyond the scope of this ticket.

> Windows is not respecting the directionality markers at all

The directionality markers are useful when the text contains a mix of LTR and RTL text, but it won't specify if the file as a whole should be displayed in RTL. If the Windows OS is set to Hebrew (for example), text in Notepad will be displayed in RTL by default (or so I've understood). Switching between RTL and LTR is manual if the text direction does not match the OS direction (this is also true for Mac).

> English shows the unicode control characters

The FSI/PDI characters can be optionally shown/hidden in Notepad with app settings, we just can't assume that the characters will be hidden by default.

> If we don't use PDFs (unsure if that would solve this)

Switching to PDF will have it's own set of problems, because then we'll have to conditionally apply styling incl. directionality markers to the PDF that we're generating on the fly, based on the user's locale (BUT at least we would have more control over the output... maybe. PDFs can also have problems with encoding http://www.rdpslides.com/psfaq/ENCODING.PDF)

> I wonder if Windows would [interpret "overrides" or "embedding" better than "isolates"](https://www.w3.org/International/questions/qa-bidi-unicode-controls.en.html#basedirection)? I'm not sure if we can parse the text and replace `[U+2068]` with `\u202E` (not 100% sure that's how it'd need to be in code) or if that would need to happen on the l10n side somehow. 🤔

I'm not sure if Windows might interpret those other characters more easily, but the trouble with "overrides" and "embedding" is that they can have spillover effect on surrounding text. The Unicode standards recommend using "isolates".